components/engine_traits/src/engine.rs

```diff
@@ -57,3 +57,11 @@ pub trait KvEngine:
     /// It cannot be used forever.
     fn bad_downcast<T: 'static>(&self) -> &T;
 }
+
+/// A factory trait to create new engine.
+///
+// It should be named as `EngineFactory` for consistency, but we are about to rename
+// engine to tablet, so always use tablet for new traits/types.
```

Comment:
I think `tablet` should replace the term —— `Engines` instead of `Engine`

Reply:
Actually, it should be
```
pub struct Engines<Tablet, RaftEngine> {
    raft_engine: RaftEngine,
    tablets: HashMap<u64, Tablet>,
}
```

Reply:
can it be like this:
```
pub struct Engines<KvEngine, RaftEngine> {
    raft_engine: RaftEngine,
    tablets: HashMap<u64, KvEngine>,
}
```
tablet is like a type name, KvEngine should be still the trait name

Reply:
Tablet is not a type name, it's an variable name. The point of renaming it is `TabletFactory` will not just act as a `KvEngine` factory, but it also manages tablets that serve the queries to tablets and control their lifetime.